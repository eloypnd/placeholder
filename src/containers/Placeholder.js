import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchPlaceholder } from '../ducks/duck'
import { PlaceholderComponent } from '../components'

export class PlaceholderContainer extends Component {
  // Lifecycle methods

  componentDidMount () {
    this.props.fetchPlaceholder()
  }

  // render

  render () {
    const { isLoading, data } = this.props
    if (isLoading) return <div>Loading...</div>
    return <PlaceholderComponent data={data} />
  }
}

const mapStateToProps = state => state.duck

const mapDispatchToProps = dispatch => ({
  fetchPlaceholder: () => dispatch(fetchPlaceholder())
})

export default connect(mapStateToProps, mapDispatchToProps)(PlaceholderContainer)
