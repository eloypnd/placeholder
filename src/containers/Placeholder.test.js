import React from 'react'
import renderer from 'react-test-renderer'
import { PlaceholderContainer } from './Placeholder'

it('PlaceholderContainer renders correctly', () => {
  const tree = renderer
    .create(<PlaceholderContainer fetchPlaceholder={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
