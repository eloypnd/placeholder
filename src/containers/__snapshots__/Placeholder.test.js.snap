// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderContainer renders correctly 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
  >
    Boilerplate
  </h1>
  <a
    href="https://en.wikipedia.org/wiki/Single-page_application"
    rel="noopener noreferrer"
    target="_blank"
  >
    SPA
  </a>
  <br />
  <small>
    with
  </small>
  <p>
    <a
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      create-react-app
    </a>
    <br />
    +
    <br />
    <a
      href="https://redux.js.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      redux
    </a>
  </p>
</div>
`;
